command/var_put.go

```diff
@@ -0,0 +1,472 @@
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/mapstructure"
+	"github.com/posener/complete"
+)
+
+type VarPutCommand struct {
+	Meta
+
+	contents  []byte
+	inFmt     string
+	outFmt    string
+	tmpl      string
+	testStdin io.Reader // for tests
+}
+
+func (c *VarPutCommand) Help() string {
+	helpText := `
+Usage: nomad var put [options] <path> [<key>=<value>]...
+
+  The 'var put' command is used to create or update an existing variable.
+  The items to be stored in the variable can be supplied as a series of
+  key-value pairs. The value for a key-value pair can be a string, an
+  @-prefixed file reference, or a '-' to get the value from standard input.
+
+  If ACLs are enabled, this command requires a token with the 'var:write'
+  capability.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault) + `
+
+Apply Options:
+  -check-index
+     If set, the variable is only purged if the server side version's modify
+     index matches the provided value.
```

Comment:
I tried to make the check-index text more portable across commands so that it could be more copypastable and I think I got to the heart of your point here. Can rework as needed.

Reply:
Looks great